pyomo/solvers/plugins/solvers/gurobi_direct.py

```diff
@@ -499,15 +514,11 @@ def _add_constraint(self, con):
         conname = self._symbol_map.getSymbol(con, self._labeler)
 
         if con._linear_canonical_form:
-            gurobi_expr, referenced_vars = self._get_expr_from_pyomo_repn(
+            gurobi_expr, referenced_vars, degree = self._get_expr_from_pyomo_repn(
                 con.canonical_form(), self._max_constraint_degree
             )
-        # elif isinstance(con, LinearCanonicalRepn):
```

Comment:
Is this commented block something that should be kept ? Happy to leave it as is to minimize the diff

Reply:
I think that is completely removable (looking back, I think `LinearCanonicalRepn` disappeared from Pyomo more than 7 years ago).